### DISCUSSION TITLE
Migrate old scripts to git

### DISCUSSION THREAD

**Discussion by 2bitninja** [community/community#69529] [Repositories] [answered]
### Select Topic Area

Question

### Body

I need some advice with migrating some old scripts, not under version control, to git. Even though the scripts where not under any type version control software. I keeped records of all the changes. When I migrate to git, it would be nice if I can show some revision history before it came under git. How do I do this?

**Comment by thevindu-w** [comment, marked as answer]
You can create a git repository and add the changes in the same order you made them since you have the records of all changes. Here are the steps,
1. Create an initially empty git repo. (Do this in an initially empty folder. Later you will add the changes)
```
git init
```

2. Add the first version of your scripts into that folder
3. Stage the changes and Create a commit (you can set the date in the past if you want)
```
git add .
git commit --date="2022.03.22 11:54" -m "Commit message"
```
4. Add the next change (from your records)
5. Stage and commit the changes (you can set the date in the past if you want)
```
git add .
git commit --date="2022.03.22 11:54" -m "Commit message"
```
6. Repeat the steps 4 and 5 until you add all the changes
7. (Optional) Create a GitHub repository and push the commits
```
git remote add origin https://github.com/<username>/<repo_name>.git
git push -u origin master master
```

**Reply by 2bitninja** [reply]
Thanks for your help.
